src/main/java/uk/gov/justice/digital/delius/controller/secure/OffendersResource.java

```diff
@@ -363,5 +364,29 @@ public Conviction getConvictionForOffenderByCrnAndConvictionId(
             .orElseThrow(() -> new NotFoundException(String.format("Offender with crn %s not found", crn)))
             .orElseThrow(() -> new NotFoundException(String.format("Conviction with ID %s for Offender with crn %s not found", convictionId, crn)));
     }
+
+    @ApiOperation(value = "Return the NSIs for a conviction ID and a CRN, filtering by NSI codes")
+    @ApiResponses(
+        value = {
+            @ApiResponse(code = 200, message = "OK", response = Conviction.class),
+            @ApiResponse(code = 400, message = "Invalid request", response = ErrorResponse.class),
+            @ApiResponse(code = 401, message = "Unauthorised", response = ErrorResponse.class),
+            @ApiResponse(code = 403, message = "Forbidden", response = ErrorResponse.class),
+            @ApiResponse(code = 404, message = "The offender CRN is not found", response = ErrorResponse.class),
+            @ApiResponse(code = 500, message = "Unrecoverable error whilst processing request.", response = ErrorResponse.class)
+        })
+    @GetMapping(path = "/offenders/crn/{crn}/convictions/{convictionId}/nsis/{nsiCodes}")
+    public NsiWrapper getNsiForOffenderByCrnAndConvictionId(
+        @ApiParam(name = "crn", value = "CRN for the offender", example = "A123456", required = true)
+        @NotNull @PathVariable(value = "crn") final String crn,
+        @ApiParam(name = "crn", value = "ID for the conviction / event", example = "2500295345", required = true)
+        @NotNull @PathVariable(value = "convictionId") final Long convictionId,
+        @ApiParam(name = "nsiCodes", value = "list of NSI codes to constrain by", example = "BRE,BRES", required = true)
+        @NotNull @PathVariable(value = "nsiCodes") final List<String> nsiCodes) {
```

Comment:
this looks odd.  Filtering would normally done by query parameters rather than path variables, since you're getting a resource at `"/offenders/crn/{crn}/convictions/{convictionId}/nsis` and then filtering

Reply:
PathVariable works and I believe offers better opportunities for cache control

Reply:
cache control?  Have we got something in front of this doing the caching?